forui/lib/src/widgets/line_calendar/line_calendar.dart

```diff
@@ -0,0 +1,346 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+
+import 'package:meta/meta.dart';
+import 'package:sugar/sugar.dart';
+
+import 'package:forui/forui.dart';
+import 'package:forui/src/widgets/line_calendar/line_calendar_tile.dart';
+
+const _textSpacing = 2.0;
+
+/// A calendar that can be scrolled horizontally.
+class FLineCalendar extends StatefulWidget {
+  /// The style. Defaults to [FThemeData.lineCalendarStyle].
+  final FLineCalendarStyle? style;
+
+  /// The controller.
+  final FCalendarController<DateTime?> controller;
+
+  /// The start date. It is truncated to the nearest date.
+  ///
+  /// ## Contract
+  /// Throws [AssertionError] if [end] <= [start].
+  final DateTime start;
+
+  /// The end date. It is truncated to the nearest date.
+  ///
+  /// ## Contract
+  /// Throws [AssertionError] if [end] <= [start].
+  final DateTime? end;
+
+  /// The current date. It is truncated to the nearest date. Defaults to the [DateTime.now].
+  final DateTime today;
+
+  /// Creates a [FLineCalendar].
+  FLineCalendar({
+    required this.controller,
+    this.style,
+    this.end,
+    DateTime? start,
+    DateTime? today,
+    super.key,
+  })  : start = start ?? DateTime(1900).toUtc(),
+        assert(
+          start == null || end == null || start.toLocalDate() < end.toLocalDate(),
+          'end date must be greater than start date',
+        ),
+        today = today ?? DateTime.now();
```

Comment:
```suggestion
        today = today ?? DateTime.now(),
        assert(
          start == null || end == null || start.toLocalDate() < end.toLocalDate(),
          'end date must be greater than start date',
        );
```

Reply:
This isn't 100% correct. DateTime.now() returns a local date time, not one in UTC.